src/overlays/actors/ovl_En_Hanabi/z_en_hanabi.c

```diff
@@ -32,38 +33,328 @@ const ActorInit En_Hanabi_InitVars = {
+        sp34.x += sp30 * Math_SinS(this->actor.home.rot.y);
+        sp34.y += randPlusMinusPoint5Scaled(300.0f);
+        sp34.z += sp30 * Math_CosS(this->actor.home.rot.y);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/func_80B22FA8.s")
+        if (this->actor.home.rot.x >= 0x259) {
+            sp28 = func_80B22C80(&sp34, this->unk_148, 2.0f);
+        } else {
+            sp28 = func_80B22C80(&sp34, this->unk_148, 1.0f);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/EnHanabi_Init.s")
+        this->actor.home.rot.y += (s16)((Rand_ZeroFloat(40.0f) + 80.0f) * 256.0f);
+        this->unk_144 = (s32)Rand_ZeroFloat(5.0f) + 20;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_OC_FIREWORKS);
+        func_80B235CC(this, &sp34, sp28);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/EnHanabi_Destroy.s")
+void func_80B23894(EnHanabi* this, GlobalContext* globalCtx) {
+    func_80B236C8(this, globalCtx);
+    if (func_80B22F34(this->unk_148) == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/func_80B234C8.s")
+void func_80B238D4(EnHanabi* this, GlobalContext* globalCtx) {
+    func_80B236C8(this, globalCtx);
+    if (this->actor.home.rot.x == 0) {
+        this->actionFunc = func_80B23894;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/func_80B235CC.s")
+void func_80B23910(EnHanabi* this, GlobalContext* globalCtx) {
+    if (this->actor.home.rot.x != 0) {
+        this->actionFunc = func_80B238D4;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/func_80B236C8.s")
+void func_80B23934(EnHanabi* this, GlobalContext* globalCtx) {
+    if ((gSaveContext.entranceIndex == 0x5410) && (gSaveContext.sceneSetupIndex == 7)) {
+        if (globalCtx->csCtx.frames > 1650) {
+            func_80B236C8(this, globalCtx);
+            func_800B8FE8(&this->actor, NA_SE_EV_FIREWORKS_LAUNCH - SFX_FLAG);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/func_80B23894.s")
+    if ((globalCtx->sceneNum == SCENE_00KEIKOKU) && (gSaveContext.sceneSetupIndex == 7) &&
+        (globalCtx->csCtx.currentCsIndex == 0) && (globalCtx->csCtx.frames == 610)) {
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_KYOJIN_GROAN);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/func_80B238D4.s")
+void EnHanabi_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnHanabi* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/func_80B23910.s")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/func_80B23934.s")
+    func_80B22E0C(this->unk_148);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hanabi/EnHanabi_Update.s")
+void func_80B23A38(Actor* thisx, GlobalContext* globalCtx) {
```

Comment:
This may not be the default Draw, but it is the only one, so just name it.
```suggestion
void EnHanabi_Draw(Actor* thisx, GlobalContext* globalCtx) {
```